### PR TITLE
Field manager namelist

### LINKED ISSUE #1313: Print message that indicates what version of the data_table and field_table is used
**Is your feature request related to a problem? Please describe.**
Currently, the stdout does not indicate whether the code is using the legacy ascii field_table/diag_table or a yaml. The only way to figure this out is find out whether the FMS was compiled with or without yaml, but soon FMS is always going to be compiled with yaml.

**Describe the solution you'd like**
Data_override and the field_manager should print a message that indicates whether you are using the legacy ascii field_table/diag_table or a yaml:
https://github.com/NOAA-GFDL/FMS/blob/0efc2e5c57d080d16fb3e076e9edc21a860992fa/data_override/data_override.F90#L243-L250

Yaml:
https://github.com/NOAA-GFDL/FMS/blob/0efc2e5c57d080d16fb3e076e9edc21a860992fa/field_manager/field_manager.F90#L551
Ascii:
https://github.com/NOAA-GFDL/FMS/blob/0efc2e5c57d080d16fb3e076e9edc21a860992fa/field_manager/field_manager.F90#L823

**Describe alternatives you've considered**
Guessing

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by uramirez8707**
**Description**
Adds a namelist to specify whether or not use the yaml tables (`use_field_table_yaml`)
The code crashes if `use_field_table_yaml = . true.` and are not compiling with yaml.
The code crashes if both versions of the tables are present
Adds units test to test all the options

Fixes #1313 

**How Has This Been Tested?**
CI

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [x] `make distcheck` passes

### CODE REVIEW

File: field_manager/field_manager.F90 [hunk 1]
```diff
@@ -560,6 +555,39 @@ subroutine field_manager_init(nfields, table_name)
 integer,                      intent(out), optional :: nfields    !< number of fields
 character(len=fm_string_len), intent(in),  optional :: table_name !< Name of the field table, default
 
+if (module_is_initialized) then
+   if(present(nfields)) nfields = num_fields
+   return
+endif
+
+call initialize
```
Comment (bensonr) [inline field_manager/field_manager.F90:563]: This name is somewhat generic, should it be made more explicit?

File: field_manager/field_manager.F90 [hunk 2]
```diff
@@ -560,6 +555,39 @@ subroutine field_manager_init(nfields, table_name)
...
+
+if (use_field_table_yaml) then
```
Comment (bensonr) [inline field_manager/field_manager.F90:565]: You have _!TODO_ comments throughout the code, but not one here to flag this for area for eventual removal.

File: test_fms/field_manager/test_field_table_read.F90 [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+use mpp_mod, only : mpp_pe, mpp_root_pe, mpp_error, NOTE, FATAL
+
+implicit none
+
+integer :: nfields
+
+call fms_init
+call field_manager_init(nfields)
+if (nfields .ne. 4) &
+  call mpp_error(FATAL, "The number fields is not the epected result")
```
Comment (bensonr) [inline test_fms/field_manager/test_field_table_read.F90:48]: While I'm being pedantic, _expected_ is misspelled.

File: field_manager/field_manager.F90 [hunk 4]
```diff
@@ -829,7 +850,7 @@ end subroutine new_name
 !! needed within various modules. The field manager does not
 !! initialize any of those schemes however. It simply holds the
 !! information and is queried by the appropriate  module.
-subroutine field_manager_init(nfields, table_name)
+subroutine field_manager_init_legacy(nfields, table_name)
```
Comment (thomas-robinson) [inline field_manager/field_manager.F90:853]: I would prefer `field_manager_init_field_table_original` or something like that.  Legacy is not descriptive enough.

File: field_manager/field_manager.F90 (hunk 1 above)
Comment (uramirez8707) [reply field_manager/field_manager.F90]: renamed to `initialize_module_variables`

File: test_fms/field_manager/test_field_table_read.F90 (hunk 3 above)
Comment (uramirez8707) [reply test_fms/field_manager/test_field_table_read.F90]: fixed

File: field_manager/field_manager.F90 (hunk 4 above)
Comment (uramirez8707) [reply field_manager/field_manager.F90]: renamed to `read_field_table_legacy` and `read_field_table_yaml`

**Review by rem1776 — APPROVED** [review]

**Review by thomas-robinson — APPROVED** [review]

**Review by bensonr — APPROVED** [review]

**Pull request merged**